src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,115 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K, V>[] map;
+    private int currentCapacity;
+    private int threshold;
 
     @Override
     public void put(K key, V value) {
-
+        updateThreshold();
+        if (size >= threshold) {
+            resize();
+        }
+        int index = generateIndex(key);
+        if (map[index] == null) {
+            map[index] = new Node<>(key, value);
+            size++;
+        }
+        if (index != 0) {
+            Node<K,V> tempNode = findInsertNode(map[index], key);
+            if (tempNode.next == null && !tempNode.key.equals(key)) {
+                tempNode.next = new Node<>(key, value);
+                size++;
+            } else {
+                tempNode.value = value;
+            }
+        } else {
+            map[index].value = value;
+        }
     }
 
     @Override
     public V getValue(K key) {
+        if (map == null) {
+            return null;
+        }
+        int index = generateIndex(key);
+        Node<K,V> tempNode = map[index];
+        if (index == 0) {
+            return map[index].value;
+        }
```

Comment:
i think you can remove those lines

Reply:
That is exception. When index is 0 (equals null in map) we may get NPE in the following code. FE map[index].key.equals(key) may oocur an exception, because map[index].key == null

Reply:
not only null key can be stored on index 0, but any other key whose hash gives 0
collision in bucket with index 0 just includes `null` key

so let's refactor your solution